src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -275,6 +278,38 @@ internal BoundExpression BindToNaturalType(BoundExpression expression, Diagnosti
                         result = ConvertSwitchExpression(expr, commonType, conversionIfTargetTyped: null, diagnostics, hasErrors);
                     }
                     break;
+                case BoundUnconvertedConditionalOperator op:
+                    {
+                        TypeSymbol type = op.Type;
+                        bool hasErrors = op.HasErrors;
+                        if (type is null)
+                        {
+                            Debug.Assert(op.NoCommonTypeError != 0);
+                            type = CreateErrorType();
+                            hasErrors = true;
+                            object trueArg = op.Consequence.Display;
+                            object falseArg = op.Alternative.Display;
+                            if (op.NoCommonTypeError == ErrorCode.ERR_InvalidQM)
+                            {
+                                if (trueArg is Symbol trueSymbol && falseArg is Symbol falseSymbol)
+                                {
+                                    SymbolDistinguisher distinguisher = new SymbolDistinguisher(this.Compilation, trueSymbol, falseSymbol);
+                                    trueArg = distinguisher.First;
+                                    falseArg = distinguisher.Second;
+                                }
+
+                                diagnostics.Add(ErrorCode.ERR_InvalidQM, op.Syntax.Location, trueArg, falseArg);
```

Comment:
>ERR_InvalidQM [](start = 58, length = 13)

I know you didn't introduce these error codes in this PR, but this is a really bad name. Consider either commenting to say what it is or changing the constant name.